wandb/sdk/artifacts/artifact.py

```diff
@@ -296,11 +302,15 @@ def _from_attrs(
         artifact._entity = entity
         artifact._project = project
         artifact._name = name
-        version_aliases = [
+        aliases = [
             alias["alias"]
-            for alias in attrs.get("aliases", [])
-            if alias["artifactCollectionName"] == name.split(":")[0]
-            and util.alias_is_version_index(alias["alias"])
+            for alias in attrs["aliases"]
+            if alias["artifactCollection"]["project"]["entityName"] == entity
+            and alias["artifactCollection"]["project"]["name"] == project
+            and alias["artifactCollection"]["name"] == name.split(":")[0]
+        ]
+        version_aliases = [
+            alias for alias in aliases if util.alias_is_version_index(alias)
```

Comment:
Question: is it possible to have `len(version_aliases) > 1`? I know creating `v.*` aliases are blocked now but are there still older artifacts out there that could have more than one version aliases? also does `artifact._version = version_aliases[0]` code below guarantee the right version?

Reply:
We currently don't allow adding vN aliases [here](https://github.com/wandb/core/blob/fdd701828b089c72597f15637c598804657b8c2d/services/gorilla/api/resolver/mutation.go#L11062-L11065). This was added in June 2020 ([PR](https://github.com/wandb/core/pull/5321)). Artifacts were added to the SDK in July 2020 ([PR](https://github.com/wandb/client-ng/pull/30)). I think it's safe to assume there are no existing artifacts that have a vN alias.

vN is added to aliases [here](https://github.com/wandb/core/blob/fdd701828b089c72597f15637c598804657b8c2d/services/gorilla/api/resolver/file.go#L1504).